crates/bevy_render/src/color/mod.rs

```diff
@@ -1332,38 +1322,45 @@ impl encase::private::CreateFrom for Color {
 
 impl encase::ShaderSize for Color {}
 
-#[derive(Debug, Error)]
+#[derive(Debug, Error, PartialEq, Eq)]
 pub enum HexColorError {
     #[error("Unexpected length of hex string")]
     Length,
-    #[error("Error parsing hex value")]
-    Hex(#[from] hex::FromHexError),
+    #[error("Invalid hex char")]
+    Char(char),
 }
 
-fn decode_rgb(data: &[u8]) -> Result<Color, HexColorError> {
-    let mut buf = [0; 3];
-    match hex::decode_to_slice(data, &mut buf) {
-        Ok(_) => {
-            let r = buf[0] as f32 / 255.0;
-            let g = buf[1] as f32 / 255.0;
-            let b = buf[2] as f32 / 255.0;
-            Ok(Color::rgb(r, g, b))
-        }
-        Err(err) => Err(HexColorError::Hex(err)),
+// Convert hex to rgb[a] u8
+// For rgb: `fff` -> [u8; 6] -> [r, g, b, ..]
+// For rgba: `E2E2E2FF` -> [u8; 8] -> [r, g, b, a, ..]
```

Comment:
```suggestion
/// Converts ASCII hex digits to an array of rgb[a] components
///
/// # Example
/// For RGB: "fff" -> `[255, 255, 255, ..]`
/// For RGBA: "E2E2E2FF" -> `[226, 226, 226, 255, ..]`
```

This really isn't sufficient. This barely explains as well as the code below does.